ibis/backends/dask/execution/util.py

```diff
@@ -355,6 +396,52 @@ def helper(
     return df
 
 
+def add_sequential_partitioned_sorted_column(
```

Comment:
Using this to partition dataframes over the previous implementation.
This is definitely less efficient and less safe from overflow, but having an index that's globally sequential is important for working with downstream operations.

Reply:
Maybe you explain this later, but can you elaborate on 

> index that's globally sequential is important for working with downstream operations.

In the prior implementation you know that the index is partitioned and what those values are.

Reply:
An example expression where this would come up is
`alltypes.int_col - alltypes.int_col.min().over(ibis.window())`

The problem shows up when we try to execute the subtraction binary op. 
`alltypes.int_col` is a dataframe with indices `[0, len(alltypes))`
`alltypes.int_col.min().over(ibis.window())` gives back a dataframe with `_ibis_index` that we set.